signer/local/local_test.go

```diff
@@ -476,8 +478,67 @@ func TestCAIssuing(t *testing.T) {
 
 const testCSR = "testdata/ecdsa256.csr"
 
+func TestPopulateSubjectFromCSR(t *testing.T) {
+	// a subject with all its fields full.
+	fullSubject := &signer.Subject{
+		CN: "CN",
+		Names: []csr.Name{
+			{
+				C:  "C",
+				ST: "ST",
+				L:  "L",
+				O:  "O",
+				OU: "OU",
+			},
+		},
+	}
+
+	fullName := pkix.Name{
+		CommonName:         "CommonName",
+		Country:            []string{"Country"},
+		Province:           []string{"Province"},
+		Organization:       []string{"Organization"},
+		OrganizationalUnit: []string{"OrganizationalUnit"},
+	}
+
+	noCN := *fullSubject
+	noCN.CN = ""
+	name := PopulateSubjectFromCSR(&noCN, fullName)
+	if name.CommonName != "CommonName" {
+		t.Fatal("Failed to replace empty common name")
+	}
+
+	noC := *fullSubject
+	noC.Names[0].C = ""
+	name = PopulateSubjectFromCSR(&noC, fullName)
+	if !reflect.DeepEqual(name.Country, fullName.Country) {
+		t.Fatal("Failed to replace empty country")
+	}
+
+	noL := *fullSubject
+	noL.Names[0].L = ""
+	name = PopulateSubjectFromCSR(&noL, fullName)
+	if !reflect.DeepEqual(name.Locality, fullName.Locality) {
+		t.Fatal("Failed to replace empty locality")
+	}
+
+	noO := *fullSubject
+	noO.Names[0].O = ""
+	name = PopulateSubjectFromCSR(&noO, fullName)
+	if !reflect.DeepEqual(name.Organization, fullName.Organization) {
+		t.Fatal("Failed to replace empty organization")
+	}
+
+	noOU := *fullSubject
+	noOU.Names[0].OU = ""
+	name = PopulateSubjectFromCSR(&noOU, fullName)
+	if !reflect.DeepEqual(name.OrganizationalUnit, fullName.OrganizationalUnit) {
+		t.Fatal("Failed to replace empty organizational unit")
+	}
+
+}
 func TestOverrideSubject(t *testing.T) {
-	csrPEM, err := ioutil.ReadFile(testCSR)
+	csrPEM, err := ioutil.ReadFile("testdata/test.csr")
```

Comment:
just a nit: can you move it to the const section and give it a name, e.g. fullsubjectCSR?